### PR TITLE
Add to_matrix_cl() vector<var> and minor OpenCL fixes

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR: 

- adds to_matrix_cl for std::vector<var_value<T>>
- fixes a header guard that caused rows(matrix_cl) to not be available
- reorder the main header file (its not opencl/prim.hpp and opencl/rev.hpp)

## Tests

Added simple test for the new function.

## Side Effects

/

## Release notes

Added to_matrix_cl(std::vector<var_value<T>>) and reorganized main Stan Math OpenCL headers.

## Checklist

- [x] Copyright holder: Rok Češnovar, Uni. of Ljubljana

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: stan/math/opencl/rev/copy.hpp [hunk 1]
```diff
@@ -53,6 +53,31 @@ inline var_value<matrix_cl<value_type_t<T>>> to_matrix_cl(
...
+ * data stored on the OpenCL device.
+ *
+ * @tparam T type of the std::vector
+ * @param a source Eigen matrix
+ * @return var with a copy of the data on the OpenCL device
+ */
+template <typename T, require_stan_scalar_t<T>* = nullptr>
+inline var_value<matrix_cl<value_type_t<T>>> to_matrix_cl(
+    std::vector<var_value<T>>& a) {
+  return new internal::op_copy_to_cl_vari<decltype(
```
Comment (t4c1) [inline stan/math/opencl/rev/copy.hpp:67]: This is not ok. The vector can get destructed before the chain is called. Instead you should just call `to_matrix_cl` (the last overload in this file) on the map. That will take care of storing the varis in the arena.

File: stan/math/prim.hpp [hunk 2]
```diff
@@ -11,4 +11,6 @@
 #include <stan/math/prim/functor.hpp>
 #include <stan/math/prim/prob.hpp>
 
+#include <stan/math/opencl/prim.hpp>
```
Comment (t4c1) [inline stan/math/prim.hpp:14]: Should this be ifdefed?

**Review by t4c1 — CHANGES_REQUESTED** [review]

File: stan/math/prim.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/prim.hpp:14]: The opencl/prim header has an ifdef on its own but I guess it does not hurt us.

File: stan/math/opencl/rev/copy.hpp (hunk 1 above)
Comment (rok-cesnovar) [reply stan/math/opencl/rev/copy.hpp]: Thanks. Done.

File: stan/math/prim.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/prim.hpp:15]: Done.

**Review by t4c1 — DISMISSED** [review]

**Comment by rok-cesnovar** [comment]
How rude of you Jenkins :)

**Review by t4c1 — APPROVED** [review]

**Pull request merged**